### ISSUE TITLE
SNOW-903710: snowpark package create includes all files and folder from teh folder the command is running in

### ISSUE THREAD

**Issue by sfc-gh-mstellwall** [Snowflake-Labs/snowcli#340]
### Python version

Python 3.10.12 (main, Jul  5 2023, 14:49:34) [Clang 14.0.6 ]

### Platform

macOS-13.5-arm64-arm-64bit

### What happened

Running `snow -v snowpark package create arch` while being in a folder that contains other files and directories will include all of those in the resulting zip.

Maybe it would be a good idea to either create a folder to download the files into as part of the command or to warn if running it in a non-empty foder.

### Console output

```Shell
2023-08-30 14:46:48 INFO 'arch' not found in Snowflake anaconda channel...
The package is not in Anaconda. Do you want to try to see if it's supported as a custom package (requires pip)? [Y/n]: Y
2023-08-30 14:46:54 INFO Collecting arch
2023-08-30 14:46:54 INFO Obtaining dependency information for arch from https://files.pythonhosted.org/packages/6e/5c/036686b39c054ce2d577c351994d7020d45cc8f32d9042c7dcd087bde416/arch-6.1.0-cp310-cp310-macosx_11_0_arm64.whl.metadata
2023-08-30 14:46:54 INFO Using cached arch-6.1.0-cp310-cp310-macosx_11_0_arm64.whl.metadata (11 kB)
2023-08-30 14:46:54 INFO Collecting numpy>=1.19 (from arch)
2023-08-30 14:46:54 INFO Obtaining dependency information for numpy>=1.19 from https://files.pythonhosted.org/packages/c3/ea/1d95b399078ecaa7b5d791e1fdbb3aee272077d9fd5fb499593c87dec5ea/numpy-1.25.2-cp310-cp310-macosx_11_0_arm64.whl.metadata
2023-08-30 14:46:54 INFO Using cached numpy-1.25.2-cp310-cp310-macosx_11_0_arm64.whl.metadata (5.6 kB)
2023-08-30 14:46:54 INFO Collecting scipy>=1.5 (from arch)
2023-08-30 14:46:54 INFO Obtaining dependency information for scipy>=1.5 from https://files.pythonhosted.org/packages/63/b9/0344b60e7c577eb637785841222fde8ef7928ec4797be1a34ca39bfe31dc/scipy-1.11.2-cp310-cp310-macosx_12_0_arm64.whl.metadata
2023-08-30 14:46:54 INFO Using cached scipy-1.11.2-cp310-cp310-macosx_12_0_arm64.whl.metadata (100 kB)
2023-08-30 14:46:54 INFO Collecting pandas>=1.1 (from arch)
2023-08-30 14:46:54 INFO Obtaining dependency information for pandas>=1.1 from https://files.pythonhosted.org/packages/4a/f6/f620ca62365d83e663a255a41b08d2fc2eaf304e0b8b21bb6d62a7390fe3/pandas-2.0.3-cp310-cp310-macosx_11_0_arm64.whl.metadata
2023-08-30 14:46:54 INFO Using cached pandas-2.0.3-cp310-cp310-macosx_11_0_arm64.whl.metadata (18 kB)
2023-08-30 14:46:54 INFO Collecting statsmodels>=0.12 (from arch)
2023-08-30 14:46:54 INFO Using cached statsmodels-0.14.0-cp310-cp310-macosx_11_0_arm64.whl (9.4 MB)
2023-08-30 14:46:54 INFO Collecting python-dateutil>=2.8.2 (from pandas>=1.1->arch)
2023-08-30 14:46:54 INFO Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
2023-08-30 14:46:55 INFO Collecting pytz>=2020.1 (from pandas>=1.1->arch)
2023-08-30 14:46:55 INFO Using cached pytz-2023.3-py2.py3-none-any.whl (502 kB)
2023-08-30 14:46:55 INFO Collecting tzdata>=2022.1 (from pandas>=1.1->arch)
2023-08-30 14:46:55 INFO Using cached tzdata-2023.3-py2.py3-none-any.whl (341 kB)
2023-08-30 14:46:55 INFO Collecting patsy>=0.5.2 (from statsmodels>=0.12->arch)
2023-08-30 14:46:55 INFO Using cached patsy-0.5.3-py2.py3-none-any.whl (233 kB)
2023-08-30 14:46:55 INFO Collecting packaging>=21.3 (from statsmodels>=0.12->arch)
2023-08-30 14:46:55 INFO Using cached packaging-23.1-py3-none-any.whl (48 kB)
2023-08-30 14:46:55 INFO Collecting six (from patsy>=0.5.2->statsmodels>=0.12->arch)
2023-08-30 14:46:55 INFO Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
2023-08-30 14:46:55 INFO Using cached arch-6.1.0-cp310-cp310-macosx_11_0_arm64.whl (862 kB)
2023-08-30 14:46:55 INFO Using cached numpy-1.25.2-cp310-cp310-macosx_11_0_arm64.whl (14.0 MB)
2023-08-30 14:46:55 INFO Using cached pandas-2.0.3-cp310-cp310-macosx_11_0_arm64.whl (10.8 MB)
2023-08-30 14:46:55 INFO Using cached scipy-1.11.2-cp310-cp310-macosx_12_0_arm64.whl (29.6 MB)
2023-08-30 14:46:55 INFO Installing collected packages: pytz, tzdata, six, packaging, numpy, scipy, python-dateutil, patsy, pandas, statsmodels, arch
2023-08-30 14:47:02 INFO Successfully installed arch-6.1.0 numpy-1.25.2 packaging-23.1 pandas-2.0.3 patsy-0.5.3 python-dateutil-2.8.2 pytz-2023.3 scipy-1.11.2 six-1.16.0 statsmodels-0.14.0 tzdata-2023.3
2023-08-30 14:47:02 INFO Checking for dependencies available in Anaconda...
2023-08-30 14:47:02 INFO Downloaded packages: dict_keys(['patsy', 'statsmodels', 'scipy', 'python-dateutil', 'pytz', 'numpy', 'six', 'tzdata', 'packaging', 'arch', 'pandas'])
2023-08-30 14:47:03 INFO 'arch' not found in Snowflake anaconda channel...
2023-08-30 14:47:03 INFO Good news! The following package dependencies can be imported directly from Anaconda, and will be excluded from the zip: [<Requirement: "patsy==0.5.3">, <Requirement: "statsmodels==0.14.0">, <Requirement: "scipy==1.11.2">, <Requirement: "python-dateutil==2.8.2">, <Requirement: "pytz==2023.3">, <Requirement: "numpy==1.25.2">, <Requirement: "six==1.16.0">, <Requirement: "tzdata==2023.3">, <Requirement: "packaging==23.1">, <Requirement: "pandas==2.0.3">]
2023-08-30 14:47:03 INFO Package patsy: deleting 2 files
2023-08-30 14:47:03 INFO Package statsmodels: deleting 2 files
2023-08-30 14:47:03 INFO Package scipy: deleting 2 files
2023-08-30 14:47:04 INFO Package python-dateutil: deleting 2 files
2023-08-30 14:47:04 INFO Package pytz: deleting 2 files
2023-08-30 14:47:04 INFO Package numpy: deleting 2 files
2023-08-30 14:47:04 INFO Package six: deleting 3 files
2023-08-30 14:47:04 INFO Package tzdata: deleting 2 files
2023-08-30 14:47:04 INFO Package packaging: deleting 2 files
2023-08-30 14:47:04 INFO Package pandas: deleting 2 files
2023-08-30 14:47:04 INFO Checking to see if packages have native libraries...
2023-08-30 14:47:04 INFO No non-supported native libraries found in packages (Good news!)...
2023-08-30 14:47:04 INFO The package arch is supported, but does depend on the following Snowflake supported native libraries. You should include the following in your packages: [<Requirement: "patsy==0.5.3">, <Requirement: "statsmodels==0.14.0">, <Requirement: "scipy==1.11.2">, <Requirement: "python-dateutil==2.8.2">, <Requirement: "pytz==2023.3">, <Requirement: "numpy==1.25.2">, <Requirement: "six==1.16.0">, <Requirement: "tzdata==2023.3">, <Requirement: "packaging==23.1">, <Requirement: "pandas==2.0.3">]

2023-08-30 14:51:26 INFO Package arch.zip created. You can now upload it to a stage (`snow package upload -f arch.zip -s packages`) and reference it in your procedure or function.
2023-08-30 14:51:26 INFO The package arch is supported, but does depend on the following Snowflake supported native libraries. You should include the following in your packages: [<Requirement: "patsy==0.5.3">, <Requirement: "statsmodels==0.14.0">, <Requirement: "scipy==1.11.2">, <Requirement: "python-dateutil==2.8.2">, <Requirement: "pytz==2023.3">, <Requirement: "numpy==1.25.2">, <Requirement: "six==1.16.0">, <Requirement: "tzdata==2023.3">, <Requirement: "packaging==23.1">, <Requirement: "pandas==2.0.3">]
```

### How to reproduce

Run `snow -v snowpark package create arch` when being a folder that has additional files and directories.

**Comment by sfc-gh-pjob** [comment]
We are currently working on a solution which will mitigate or eliminate this issue. We'll update this issue as soon as we finish the changes and we'll let you know in which release the changes should be expected.

**Comment by sfc-gh-turbaszek** [comment]
It's been fixed in #561

**Comment by sfc-gh-turbaszek** [comment]
It's been fixed in #561
